edgelesssys/constellation
# Add pod disruption budgets so the cluster-autoscaler is able to move kube-admin namespaced resources

### Context


### Proposed change(s)

- Refactor the cilium helm values so that every CSP is based off of shared config and only overwrites CSP specific values.
- Enable the pod disruption budget for cert-manafger components 
- Enable the pod disruption budget for the cilium operator
- Manually add a podDisruptionBudget for coreDNS if the cluster-autoscaler if deployed

How to test:
* Start a Constellation cluster with initially 1 control-plane and 2 worker nodes
* Wait for all nodes to be up
* Restart the cert-manager components to make sure that some of them are scheduled on worker nodes
* Also make sure that a coreDNS pod is scheduled on a worker node
* Manually increase the control-plane count to 3 by increasing the count in the CSP console
* Wait for the other 2 control-plane node to join (Those nodes are joined to that the autoscaler has enough nodes to re-schedule all the e.g. cert-manager pods)
* Enable autoscaling and set the `min` value to `0` in the worker scaling group CR
* Have a look at the logs of the cluster-autoscaler
* After maximum of 25 minutes both worker nodes should be scaled down

### Additional info

- [AB#3692](https://dev.azure.com/Edgeless/ae37573d-ccde-4af2-ab1e-001e587197d1/_workitems/edit/3692)
- https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#what-types-of-pods-can-prevent-ca-from-removing-a-node

### Checklist
